src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LENGTH = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        values = (V[])new Object[ARRAY_LENGTH];
+        keys = (K[]) new Object[ARRAY_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
```

Comment:
```suggestion
    public void put(K key, V value) {
        if (size == ARRAY_LENGTH) {
                throw new ArrayIndexOutOfBoundsException("Storage is full");
            }
```

You should check first